kernel/src/process.rs

```diff
@@ -539,7 +540,7 @@ pub trait Process {
     ///   process's memory region.
     /// - [`Error::KernelError`] if there was an internal kernel error. This is
     ///   a bug.
-    fn brk(&self, new_break: *const u8) -> Result<*const u8, Error>;
+    fn brk(&self, new_break: *const u8) -> Result<CapabilityPtr, Error>;
```

Comment:
The argument is a `*const u8` because userspace is requesting access to memory, but fabricating an address ex nihilo. It returns a `CapabilityPtr` upon successful because the kernel has authorized access.

Reply:
Yes, this was the motivation.

Reply:
I'm confused -- the documentation says that this returns the "previous" break address. How does that related to the capability returned here? Why would we want to issue a capability (potentially with authority) over a previous address?

Reply:
@lschuermann you're right, I mispoke, but in any case, the returned value has some validity for userspace. Assuming the new break increases the heap, the returned value would be the same as the for `sbrk` (see below).

If it decreases the heap it's maybe less clear, although in either case, this documentation actually differs from what libc expects (which is the returned value to signal success, not the old pointer).